TileStache/Goodies/VecTiles/transform.py

```diff
@@ -503,37 +550,40 @@ def cut(self, shape, props, fid):
                 if cutting_shape.intersects(shape):
                     shape = self._intersect(
                         shape, props, fid, cutting_shape,
-                        cutting_attr, original_geom_type)
+                        cutting_attr, original_geom_dim)
 
-            # if there's no geometry left outside the
-            # shape, then we can exit the loop early, as
-            # nothing else will intersect.
-            if shape.is_empty:
-                break
+                # if there's no geometry left outside the
+                # shape, then we can exit the function
+                # early, as nothing else will intersect.
+                if shape.is_empty:
+                    return
 
         # if there's still geometry left outside, then it
         # keeps the old, unaltered properties.
-        self._add(shape, props, fid, original_geom_type)
+        self._add(shape, props, fid, original_geom_dim)
 
 
     # only keep geometries where either the type is the
     # same as the original, or we're not trying to keep the
     # same type.
-    def _add(self, shape, props, fid, original_geom_type):
-        if (not shape.is_empty and
-            (not self.keep_geom_type or
-             isinstance(shape, original_geom_type))):
+    def _add(self, shape, props, fid, original_geom_dim):
+        # use a custom dimension measurement here, as it
+        # turns out shapely geometry objects don't always
+        # form a hierarchy that's usable with isinstance.
+        shape_dim = _geom_dimensions(shape)
+
+        # don't add empty shapes, they're completely
+        # useless.
+        if shape.is_empty:
+            pass
```

Comment:
Maybe do this check first, and `return` instead of `pass` (if you think that is clearer)